error[E0554]: #![feature] may not be used on the stable release channel

When compiling [hypercore](https://github.com/datrs/hypercore) I get the following error:

```
error[E0554]: #![feature] may not be used on the stable release channel
 --> /Users/bryce/.cargo/registry/src/github.com-1ecc6299db9ec823/flat-tree-3.3.0/src/lib.rs:2:1
  |
2 | #![feature(external_doc)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^
```

```
cargo --version
cargo 1.28.0 (96a2c7d16 2018-07-13)
```

I'm not very familiar with Rust, is it implied that I should not be using the stable release channel?

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -1,6 +1,6 @@
-#![deny(missing_docs)]
-#![feature(external_doc)]
-#![doc(include = "../README.md")]
+#![cfg_attr(feature = "nightly", deny(missing_docs))]
+#![cfg_attr(feature = "nightly", feature(external_doc))]
+#![cfg_attr(feature = "nightly", doc(include = "../README.md"))]
 
 mod iterator;
 

